lib/Net/SIP/Dispatcher.pm

```diff
@@ -861,24 +861,36 @@ sub __resolve_uri_final {
     # FIXME: can contradict order in @proto
     @$resp = sort { $a->{prio} <=> $b->{prio} } @$resp;
 
+    # remove duplicate entries in response
+    my %resp;
+    for(my $i=0; $i<@$resp; $i++) {
+	my $r = $resp->[$i];
+	if (exists $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}}) {
```

Comment:
Please use a simpler `$resp{ @{$r}{qw( host family proto addr port)} }`  here instead of a deeply nested hash. Also, using both `%resp` and `@resp` is confusing, better use a different name more clearly associated with the task.

Reply:
I'm not sure what do you mean by usage `$resp{ @{$r}{qw( host family proto addr port)} }`. This construction in perl do same thing as `$resp{$r->{port}}`? (`$resp{@list}` would return value from hash `$resp` for key which name is the last element of `@list`)

Anyway, why I chose this nested hash is because one response is tuple and to filter duplicate responces it is needed to filter duplicate tuples. Therefore I stored whole tuple into nested hash. I do not think that perl has some Core module with function which can do `uniq` with own comparator (to define if two members are same or not). And via nested hash `$h->{val1}->{val2}->{val3}->...->{valN}` is possible to easily represent (ordered) tuple `(val1, val2, val3, ... valN)` and easily eliminate duplicates as perl hash is not multivalue hash.

Anyway, if you have better idea how to filter duplicate elements from tuple, please let me know.